imap_processing/codice/codice_l2.py

```diff
@@ -412,38 +628,28 @@ def process_codice_l2(
     l2_dataset : xarray.Dataset
         The``xarray`` dataset containing the science data and supporting metadata.
     """
-    logger.info(f"Processing {file_path}")
-
-    # Open the l1 file
-    l1_dataset = load_cdf(file_path)
+    # This should get science files since ancillary or spice doesn't have data_type
+    # as data level.
+    file_path = dependencies.get_file_paths(descriptor=descriptor)[0]
 
-    # Use the logical source as a way to distinguish between data products and
-    # set some useful distinguishing variables
-    # TODO: Could clean this up by using imap-data-access methods?
-    dataset_name = l1_dataset.attrs["Logical_source"]
-    data_level = dataset_name.removeprefix("imap_codice_").split("_")[0]
-    dataset_name = dataset_name.replace(data_level, "l2")
-
-    # Use the L1 data product as a starting point for L2
-    l2_dataset = l1_dataset.copy()
-    # Get the L2 CDF attributes
-    # cdf_attrs = ImapCdfAttributes()
-
-    # TODO uncomment and update variable attrs
-    # l2_dataset = add_dataset_attributes(l2_dataset, dataset_name, cdf_attrs)
+    # Now form product name from descriptor
+    descriptor = ScienceFilePath(file_path).descriptor
+    dataset_name = f"imap_codice_l2_{descriptor}"
 
     # TODO: update list of datasets that need geometric factors (if needed)
     # Compute geometric factors needed for intensity calculations
     if dataset_name in [
         "imap_codice_l2_lo-sw-species",
         "imap_codice_l2_lo-nsw-species",
     ]:
+        l1_dataset = load_cdf(file_path)
+        l2_dataset = l1_dataset.copy()
```

Comment:
Do you use the l1_dataset? If not, maybe just point directly to the copy. Why do you need a copy too, if you are just loading it you can be assured it is your original.
```suggestion
        l2_dataset = load_cdf(file_path).copy()
```